src/AmbientSounds.Uwp/Views/MainPage.xaml

```diff
@@ -6,12 +6,19 @@
     xmlns:controls="using:AmbientSounds.Controls"
     xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
     xmlns:extensions="using:Microsoft.Toolkit.Uwp.UI.Extensions"
-    xmlns:local="using:AmbientSounds.Views"
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
     xmlns:muxc="using:Microsoft.UI.Xaml.Controls"
     xmlns:viewmodels="using:AmbientSounds.ViewModels"
+    xmlns:interactivity="using:Microsoft.Xaml.Interactivity"
+    xmlns:core="using:Microsoft.Xaml.Interactions.Core"
     Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
     mc:Ignorable="d">
+    <interactivity:Interaction.Behaviors>
+        <core:EventTriggerBehavior EventName="Loaded">
+            <core:InvokeCommandAction Command="{x:Bind ViewModel.LoadCommand}"/>
```

Comment:
Thinking about code readability and simplicity. I'm not used to this behaviours approach, and it took me a while to read it. What's the motivation to use this approach instead of something like `Loaded="{x:Bind PageLoaded}"` where `PageLoaded` is a method that runs `ViewModel.LoadAsync()`? Is the goal just to have no code in `xaml.cs`? Are there performance reasons?

Reply:
There's a number of reasons for this, yeah:

- Binding directly to a method only works on UWP, so you'd still need the commands for other platforms
- Commands provide you with a small abstraction layer, which helps for maintainability
- You can't bind methods that return a task, which means you'd need to make the method `async void`. That's a big no-no as an exception there would just crash the whole application without the ability to handle that properly.
- Using a command also gives you extra features, eg. here I'm using an `AsyncRelayCommand`, which means you'd also easily be able to eg. bind a loading indicator to the loading method, if you wanted to.